src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -873,6 +873,33 @@
     (SSLUtils/caCertAndCrlPemsToSSLContext ca-cert-reader
                                                        crls-reader)))
 
+(defn generate-ssl-context
+  [options]
```

Comment:
@fpringvaldsen I don't know the use-case for this new function so perhaps it doesn't make sense, but so far we're writing everything in Java and then just creating Clojure wrappers. This allows us to use the functionality in Ruby (via JRuby) much easier than having to somehow go through Clojure.

Tests are still in Clojure (and testing the Clojure API, not Java directly) though. Unless there's a reason not to, I would port this to Java just to follow the convention.

And yes, Javadocs and docstrings on the static method and function would be awesome!